File: ios/MyTwitter/Models/Member.swift
import Foundation
import FirebaseFirestore

struct Member: Codable, Identifiable, Hashable {
    @DocumentID var id: String?
    var handle: String?
    var name: String?
    var avatar: String?
    var role: String?
    var enabled: Bool?
    var xUserId: String?

    var isAdmin: Bool { role == "admin" }
    var isEnabled: Bool { enabled != false }

    var displayHandle: String {
        let h = (handle ?? "").trimmingCharacters(in: .whitespacesAndNewlines)
        return h.isEmpty ? (id ?? "") : h
    }
}

File: android/app/src/main/java/org/evergreenlabs/mytwitter/data/Member.kt
package org.evergreenlabs.mytwitter.data

import com.google.firebase.firestore.DocumentId
import com.google.firebase.firestore.Exclude
import com.google.firebase.firestore.IgnoreExtraProperties

@IgnoreExtraProperties
data class Member(
    @DocumentId val id: String? = null,
    val handle: String? = null,
    val name: String? = null,
    val avatar: String? = null,
    val role: String? = null,
    val enabled: Boolean? = null,
    val xUserId: String? = null,
) {
    @get:Exclude
    val isAdmin: Boolean get() = role == "admin"

    /** Absent field counts as enabled. */
    @get:Exclude
    val isEnabled: Boolean get() = enabled != false

    @get:Exclude
    val displayHandle: String
        get() {
            val h = handle?.trim().orEmpty()
            return h.ifEmpty { id.orEmpty() }
        }
}
